### ISSUE TITLE
[FUZZ] Beaconfuzz_v2 crash-157110fa08526632a192ac5304f0b5350117fa87 in struct_block_header

### ISSUE THREAD

**Issue by Daft-Wullie** [sigp/beacon-fuzz#87]
## I've done and provided the following:
- [x] Checked to see if any other `[FUZZ]` issue already refers to that crasher
- [x] Attached the crashing input (either attached to the issue as a .zip or .gz, or as a link to a file sharing service)
- [x] Noted the `beacon-fuzz` version or commit used.
- [x] Provided crash output
- [x] Noted the command or fuzzer used to generate the crash
- [x] Name of the original crash file
- [x] (Optional but optimal) Checked if the crash can be consistently replicated by re-running the input. 
## Info to Reproduce
* Command run: e.g. `ETH2FUZZ_BEACONSTATE=../eth2fuzz/workspace/corpora/beaconstate cargo +nightly fuzz run struct_block_header fuzz/artifacts/struct_block_header/crash-157110fa08526632a192ac5304f0b5350117fa87 -- -detect_leaks=0 -rss_limit_mb=0`
* Crasher file name: crash-157110fa08526632a192ac5304f0b5350117fa87
[crash-157110fa08526632a192ac5304f0b5350117fa87.zip](https://github.com/sigp/beacon-fuzz/files/5402060/crash-157110fa08526632a192ac5304f0b5350117fa87.zip)

* Client exercised: Teku
* Fuzzing engine used (if applicable): libfuzzer

## Crash output and stacktrace

```
Picked up JAVA_TOOL_OPTIONS: -Xmx6g
Uncaught Java exception while calling target:
Exception in thread "main" org.apache.tuweni.ssz.InvalidSSZTypeException: SSZ encoded data is not a fixed-length byte array
        at org.apache.tuweni.ssz.BytesSSZReader.ensureBytes(BytesSSZReader.java:217)
        at org.apache.tuweni.ssz.BytesSSZReader.readFixedBytes(BytesSSZReader.java:61)
        at org.apache.tuweni.ssz.SSZReader.readFixedBytes(SSZReader.java:66)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializePrimitive(SimpleOffsetSerializer.java:492)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeFixedParts(SimpleOffsetSerializer.java:311)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeContainer(SimpleOffsetSerializer.java:259)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeFixedContainer(SimpleOffsetSerializer.java:287)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeFixedParts(SimpleOffsetSerializer.java:313)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeContainer(SimpleOffsetSerializer.java:259)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeFixedContainer(SimpleOffsetSerializer.java:287)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeFixedParts(SimpleOffsetSerializer.java:313)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeContainer(SimpleOffsetSerializer.java:259)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeVariableParts(SimpleOffsetSerializer.java:381)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeContainer(SimpleOffsetSerializer.java:263)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeVariableElementList(SimpleOffsetSerializer.java:437)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeVariableParts(SimpleOffsetSerializer.java:369)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeContainer(SimpleOffsetSerializer.java:263)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeVariableParts(SimpleOffsetSerializer.java:381)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeContainer(SimpleOffsetSerializer.java:263)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeVariableParts(SimpleOffsetSerializer.java:381)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeContainer(SimpleOffsetSerializer.java:263)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserializeContainerErrorWrapper(SimpleOffsetSerializer.java:239)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.lambda$deserialize$1(SimpleOffsetSerializer.java:211)
        at org.apache.tuweni.ssz.SSZ.decode(SSZ.java:1255)
        at tech.pegasys.teku.datastructures.util.SimpleOffsetSerializer.deserialize(SimpleOffsetSerializer.java:207)
        at tech.pegasys.teku.fuzz.FuzzUtil.deserialize(FuzzUtil.java:260)
        at tech.pegasys.teku.fuzz.FuzzUtil.fuzzBlockHeader(FuzzUtil.java:151)
Traceback (most recent call last, using override)
/home/nim-beacon-chain/vendor/nimbus-build-system/vendor/Nim/lib/system/excpt.nim(614) signalHandler
SIGABRT: Abnormal termination.
==885353== ERROR: libFuzzer: fuzz target exited
    #0 0x5564535ae901  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0xbd2901)
    #1 0x556455b50d80  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0x3174d80)
    #2 0x556455b65aeb  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0x3189aeb)
    #3 0x7f8f974a2a26  (/lib/x86_64-linux-gnu/libc.so.6+0x49a26)
    #4 0x7f8f974a2bdf  (/lib/x86_64-linux-gnu/libc.so.6+0x49bdf)
    #5 0x55645391a2cc  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0xf3e2cc)
    #6 0x7f8f9749f20f  (/lib/x86_64-linux-gnu/libc.so.6+0x4620f)
    #7 0x7f8f9749f18a  (/lib/x86_64-linux-gnu/libc.so.6+0x4618a)
    #8 0x7f8f9747e858  (/lib/x86_64-linux-gnu/libc.so.6+0x25858)
    #9 0x55645380b791  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0xe2f791)
    #10 0x5564538005d0  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0xe245d0)
    #11 0x5564537fc3f6  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0xe203f6)
    #12 0x5564536f3779  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0xd17779)
    #13 0x55645367eef2  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0xca2ef2)
    #14 0x556455b442f0  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0x31682f0)
    #15 0x556455b43f4f  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0x3167f4f)
    #16 0x556455b65f4c  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0x3189f4c)
    #17 0x556455b6e100  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0x3192100)
    #18 0x556455b6f2db  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0x31932db)
    #19 0x556455b70d4c  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0x3194d4c)
    #20 0x556455b42499  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0x3166499)
    #21 0x55645352b4b6  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0xb4f4b6)
    #22 0x7f8f974800b2  (/lib/x86_64-linux-gnu/libc.so.6+0x270b2)
    #23 0x55645352b65d  (/home/beacon-fuzz/beaconfuzz_v2/fuzz/target/x86_64-unknown-linux-gnu/release/struct_block_header+0xb4f65d)

SUMMARY: libFuzzer: fuzz target exited
MS: 0 ; base unit: 0000000000000000000000000000000000000000
0x9f,0x0,0x0,0x0,0x0,0x0,0x0,0x0,0x8c,0x0,0x0,0x0,0x0,0x0,0x0,0x0,0x0,0x0,0x0,0x0,0x8c,0x8c,0x8c,0x2e,0x6c,0x6f,0x6f,0x8c,0x8c,0x8c,0x8c,0x8c,0x8c,0x0,0x0,0x0,0x38,0xbf,0x8c,0x8c,0x0,0x0,0x0,0x0,0x19,0x0,0x2c,0x0,0xbf,0x8c,0x8c,0x2e,0x0,0x0,0x0,0x0,
\x9f\x00\x00\x00\x00\x00\x00\x00\x8c\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x8c\x8c\x8c.loo\x8c\x8c\x8c\x8c\x8c\x8c\x00\x00\x008\xbf\x8c\x8c\x00\x00\x00\x00\x19\x00,\x00\xbf\x8c\x8c.\x00\x00\x00\x00
artifact_prefix='/home/beacon-fuzz/beaconfuzz_v2/fuzz/artifacts/struct_block_header/'; Test unit written to /home/beacon-fuzz/beaconfuzz_v2/fuzz/artifacts/struct_block_header/crash-157110fa08526632a192ac5304f0b5350117fa87
Base64: nwAAAAAAAACMAAAAAAAAAAAAAACMjIwubG9vjIyMjIyMAAAAOL+MjAAAAAAZACwAv4yMLgAAAAA=
```

## Your Environment
* Fuzzer ran: beaconfuzz_v2
* Version/Commit used: f23ec4e
* Operating System and version: Ubuntu 20.04

**Comment by pventuzelo** [comment]
I'm able to reproduce.

@gnattishness Is InvalidSSZTypeException should be catched by the jni library?

**Comment by gnattishness** [comment]
> I'm able to reproduce.
> 
> @gnattishness Is InvalidSSZTypeException should not be catched by the jni library?

Agreed probably an issue with the fuzzer.
The `teku` FuzzUtil class should not be catching InvalidSSZTypeException, as this indicates the data passed to it is not in an expected SSZ format (but the fuzzer should be passing it validly-encoded ssz)

I'll have to confirm, but have a hunch it might be something to do with `SignedBeaconBlock` vs `BeaconBlock` data types.

**Comment by pventuzelo** [comment]
It was a bug with the fuzzer data type for teku
Fixed in 232e51b54fa4f5cff0d0e14592828e7300618251
